build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/features/BadgesDetails.java

```diff
@@ -0,0 +1,91 @@
+package com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.features;
+
+import com.google.common.collect.ImmutableList;
+import com.smartcodeltd.jenkinsci.plugins.buildmonitor.viewmodel.JobView;
+
+import hudson.model.BuildBadgeAction;
+import jenkins.model.Jenkins;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+import org.apache.commons.jelly.JellyContext;
+import org.apache.commons.jelly.JellyException;
+import org.apache.commons.jelly.XMLOutput;
+import org.codehaus.jackson.annotate.JsonValue;
+import org.kohsuke.stapler.MetaClass;
+import org.kohsuke.stapler.WebApp;
+import org.kohsuke.stapler.jelly.JellyClassLoaderTearOff;
+
+/**
+ * @author Jan Molak
+ */
+public class BadgesDetails implements Feature<BadgesDetails.Badges> {
+    private JobView job;
+
+    public BadgesDetails() {
+
+    }
+
+    @Override
+    public BadgesDetails of(JobView jobView) {
+        this.job = jobView;
+
+        return this;
+    }
+
+    @Override
+    public Badges asJson() {
+        return !job.lastCompletedBuild().badgeActions().isEmpty()                  // would be nice to have .map(Claim(_)).orElse(), but hey...
+                ? new Badges(job.lastCompletedBuild().badgeActions())
+                : null;                             // `null` because we don't want to serialise an empty object
+    }
+
+    public static class Badges {
+
+        private final List<String> badges = newArrayList();
+
+        public Badges(List<BuildBadgeAction> badgeActions) {
+        	MetaClass mc = WebApp.getCurrent().getMetaClass(getClass());
+    		JellyContext context = mc.classLoader.loadTearOff(JellyClassLoaderTearOff.class).createContext();
+
+    		context.setVariable("app", Jenkins.getInstance());
+    		context.setVariable("rootURL", Jenkins.getInstance().getRootUrl());
+    		context.setVariable("h", new hudson.Functions());
+    		context.setVariable("resURL", Jenkins.RESOURCE_PATH);
+    		context.setVariable("imagesURL", Jenkins.RESOURCE_PATH + "/images");
```

Comment:
Hmm, maybe the `StaticJenkinsAPIs` should be capable of creating the `JellyContext` to move this responsibility out of this class?